### ISSUE TITLE
Integration with react navigation

### ISSUE THREAD

**Issue by andreialecu** [PedroBern/react-native-collapsible-tab-view#7]
Thank you for creating this!

I saw that you mentioned in https://github.com/satya164/react-native-tab-view/pull/1096#issue-520914860 that there would be integration with `react-navigation`. 

I'm really interested in this, and would love to help by testing it. Alternatively, if you can share the private implementation you mentioned, I might be able to implement and test that as well, before a polished public release.

**Comment by PedroBern** [comment]
Hi @andreialecu, I don't plan to add the integration by myself, because for now it solves my use case. But if you or someone is willing to open a PR, I would accept it :)

## Implementation

The implementation is super simple, you can do it in 3 steps:

### 1. MaterialTopTabsCollapsibleTabView

Same as the [original](https://github.com/react-navigation/react-navigation/blob/main/packages/material-top-tabs/src/views/MaterialTopTabView.tsx), replacing the `TabView` [here](https://github.com/react-navigation/react-navigation/blob/main/packages/material-top-tabs/src/views/MaterialTopTabView.tsx#L51) by the `CollapsibleTabView`.

### 2. MaterialTopTabsCollapsibleNavigator

Just like in step 1, same as the [original](https://github.com/react-navigation/react-navigation/blob/main/packages/material-top-tabs/src/navigators/createMaterialTopTabNavigator.tsx), replacing the `MaterialTopTabView` [here](https://github.com/react-navigation/react-navigation/blob/main/packages/material-top-tabs/src/navigators/createMaterialTopTabNavigator.tsx#L12) by the component of step 1.

### 3. Changes in the CollapsibleTabView

There is this one caveat, currently, the `CollapsibleTabView` uses the `route.key` as identifier, and to make this work with the `react-navigation` we need to use `route.name`. So you need to make changes [here](https://github.com/PedroBern/react-native-collapsible-tab-view/blob/main/src/CollapsibleTabView.tsx#L292), and remember to use the `name` in the hooks:

```diff
--- activeRouteKey: routes[index].key,
+++ activeRouteKey: routes[index].name,
```

## Investigation

I'm not using it and it may have bugs, especially for the snap effect. Please if you can fork, experiment and maybe find ways to improve, would be really nice :)

**Comment by andreialecu** [comment]
Thanks for the pointers! I opened a PR at #10 which implements this, and it seems to work perfectly in conjunction with #9

**Comment by PedroBern** [comment]
merged #10 👍

**Comment by PedroBern** [comment]
released in [v1.2.0](https://github.com/PedroBern/react-native-collapsible-tab-view/blob/main/CHANGELOG.md#120-2020-12-02)
